cmd/ctr/commands/shim/shim.go

```diff
@@ -226,20 +228,30 @@ var execCommand = cli.Command{
 }
 
 func getTaskService(context *cli.Context) (task.TaskService, error) {
-	bindSocket := context.GlobalString("socket")
-	if bindSocket == "" {
-		return nil, errors.New("socket path must be specified")
+	id := context.GlobalString("id")
+	if id == "" {
+		return nil, fmt.Errorf("container id must be specified")
 	}
+	ns := context.GlobalString("namespace")
 
-	conn, err := net.Dial("unix", "\x00"+bindSocket)
-	if err != nil {
-		return nil, err
-	}
+	// /containerd-shim/ns/id/shim.sock is the old way to generate shim socket,
+	// compatible it
+	s1 := filepath.Join(string(filepath.Separator), "containerd-shim", ns, id, "shim.sock")
+
+	d := sha256.Sum256([]byte(filepath.Join(ns, id)))
+	s2 := filepath.Join(string(filepath.Separator), "containerd-shim", fmt.Sprintf("%x.sock", d))
```

Comment:
Can this be just `shim.SocketAddress(ctx, id)` ? https://github.com/containerd/containerd/blob/cd5369bace3f5208dadd870a7eea405905e91806/runtime/v2/shim/util_unix.go#L56

Reply:
Update.